src/analyze/analyze.c

```diff
@@ -214,13 +216,93 @@ static int compare_unit_start(const UnitTimes *a, const UnitTimes *b) {
         return CMP(a->activating, b->activating);
 }
 
+static int process_aliases(char *argv[], char *tempdir, char ***ret) {
+        int r, i = 1;
+        char **filename;
+        _cleanup_strv_free_ char **filenames = NULL;
+        struct stat statbuf;
+
+        assert(argv);
+        assert(tempdir);
+        assert(ret);
+
+        STRV_FOREACH(filename, strv_skip(argv, 1)) {
+                _cleanup_free_ char *src = NULL, *dst = NULL, *argname = NULL;
+
+                if (!strchr(*filename, ':')) {
+                        r = strv_extend(&filenames, argv[i++]);
+                        if (r < 0)
+                                return log_oom();
+
+                        continue;
+                }
+
+                argname = strdup(*filename);
+                r = extract_first_word((const char **)filename, &src, ":", 0);
+                if (r < 0 || !*filename)
+                        return log_error_errno(r, "Failed to parse alias %s", argname);
+
+                dst = path_join(tempdir, basename(*filename));
+                if (!dst)
+                        return log_oom();
+
+                r = stat(src, &statbuf);
+                if (r < 0)
+                        return log_error_errno(errno, "Couldn't stat %s", src);
+
+                r = copy_file(src, dst, 0, statbuf.st_mode, 0, 0, COPY_REFLINK);
+                if (r < 0)
+                        return log_error_errno(r, "Couldn't copy %s to alias %s", src, dst);
+
+                r = strv_consume(&filenames, TAKE_PTR(dst));
+                if (r < 0)
+                        return log_oom();
+                i++;
+        }
+
+        *ret = TAKE_PTR(filenames);
+        return 0;
+}
+
+static int setup_working_dir(char **ret) {
+        int r;
+        char *tempdir = NULL;
+        char bid[SD_ID128_STRING_MAX];
+        sd_id128_t boot_id;
+
+        assert(ret);
+
+        r = sd_id128_get_boot(&boot_id);
+        if (r < 0)
+                return log_error_errno(r, "Failed to obtain boot ID");
+
+        tempdir = strjoin("/tmp/systemd-analyze-", sd_id128_to_string(boot_id, bid), "-XXXXXX");
+        if (!tempdir)
+                return log_oom();
+
+        tempdir = mkdtemp(tempdir);
```

Comment:
`tempdir` is leaked if this fails - don't overwrite the pointer, instead simply check if mkdtemp is returning NULL or not